### PR TITLE
1434/listing finder q2

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #1434 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR refines the way the form data is handled based on if users are selecting previous, next, or skip to listings. Also, this PR includes an empty disclaimer page as a placeholder since it helped to emulate the data saving logic that the complete form requires.
 
## How Can This Be Tested/Reviewed?

This can be tested by going to the Listings Finder from the homepage by selecting "Find Listings for You". Then click through the form using next and previous to observe how selections persist after selecting next on a question, but will not if previous is clicked after the selections are made. Additionally, ensure that clicking next after the second question or the skip link takes you to the placeholder page where submit will display the listings page with the selected filters.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/pages/finder.tsx [hunk 1]
```diff
@@ -56,22 +53,53 @@ const Finder = () => {
...
+
+  const onSubmit = () => {
+    const formSelections = {}
+    formData?.forEach((question) => {
+      formSelections[question.fieldGroupName] = question?.fields
+        ?.filter((field) => field.selected)
+        ?.map((field) => field.label)
+        ?.join()
+    })
+    console.log(formSelections)
```
Comment (ludtkemorgan) [inline sites/public/pages/finder.tsx:66]: 🧹

File: sites/public/pages/finder.tsx [hunk 2]
```diff
@@ -56,22 +53,53 @@ const Finder = () => {
...
             selected: false,
           }))
-          formQuestions[1] = { fieldGroupName: "bedRoomSize", fields: bedroomFields }
+          formQuestions.push({
+            formSection: t("finder.progress.housingLabel"),
+            fieldGroupName: "bedRoomSize",
+            fields: bedroomFields,
+          })
         }
+        console.log(Region)
```
Comment (ludtkemorgan) [inline sites/public/pages/finder.tsx:88]: 🧹

File: sites/public/pages/finder.tsx [hunk 3]
```diff
@@ -56,22 +53,53 @@ const Finder = () => {
...
+        const neihborhoodFields = Object.keys(Region).map((key) => ({
```
Comment (ludtkemorgan) [inline sites/public/pages/finder.tsx:89]: small, but a typo in neighborhood

**Review by ludtkemorgan — APPROVED** [review]
This looks good. I'm assuming you are aware the "Select all that apply" is being shown on the disclaimer placeholder page and that will be tackled in the disclaimer story?

**Comment by ColinBuyck** [comment]
@ludtkemorgan Just cleaned things up and you're correct, I'll be handling that as part of the next ticket! Thank you for the review 🌟

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
A few functional notes:
* `Skip this and show me listing` should have a pointer cursor
* If I skip Q2 then return and try to fill it out, it doesn't retain the data (below)
https://user-images.githubusercontent.com/65367387/192630624-8c2aa628-d1ce-462c-92ec-2006d01609c6.mov
* It looks like `Greater Downtown` while added as a URL param isn't being auto-checked in the filter drawer (below)
* 
Screen Shot 2022-09-27 at 4 37 01 PM

File: sites/public/pages/finder.tsx [hunk 4]
```diff
@@ -22,25 +22,22 @@ import Layout from "../layouts/application"
 
 interface finderField {
```
Comment (emilyjablonski) [inline sites/public/pages/finder.tsx:23]: Nit: do you mind capitalizing these interfaces?

File: sites/public/pages/finder.tsx [hunk 5]
```diff
@@ -56,22 +53,51 @@ const Finder = () => {
...
+          label: key,
+          selected: false,
+        }))
+        formQuestions.push({
+          formSection: t("finder.progress.housingLabel"),
+          fieldGroupName: "region",
+          fields: neighborhoodFields,
+        })
+        formQuestions.push({
+          formSection: t("finder.progress.builingLabel"),
```
Comment (emilyjablonski) [inline sites/public/pages/finder.tsx:97]: `builing` --> `building`

File: sites/public/pages/finder.tsx [hunk 6]
```diff
@@ -89,15 +115,15 @@ const Finder = () => {
...
+            currentStep={sectionNumber}
             totalSteps={3}
             stepPreposition={t("finder.progress.stepPreposition")}
             stepLabeling={stepLabels}
           ></StepHeader>
         </div>
         <ProgressNav
-          currentPageSection={currentStep}
-          completedSections={currentStep - 1}
+          currentPageSection={sectionNumber}
```
Comment (emilyjablonski) [inline sites/public/pages/finder.tsx:125]: Sort of an aside, but it seems like maybe the progress nav doesn't need to keep track of both of these numbers if `completedSections` is always one less than `currentPageSection`?

File: sites/public/pages/finder.tsx [hunk 7]
```diff
@@ -152,14 +176,21 @@ const Finder = () => {
                             key={FrontendListingFilterStateKeys[field.label]}
                           >
                             <Field
-                              name="bedRoomSize"
+                              name={activeQuestion.fieldGroupName}
                               register={register}
                               id={FrontendListingFilterStateKeys[field.label]}
-                              label={t(`listingFilters.bedroomsOptions.${field.translation}`)}
+                              label={
+                                field.translation
```
Comment (emilyjablonski) [inline sites/public/pages/finder.tsx:183]: What are the cases where we don't want the translation?

File: sites/public/pages/finder.tsx [hunk 8]
```diff
@@ -172,10 +203,12 @@ const Finder = () => {
                 </div>
 
                 <div className="bg-gray-300 flex flex-row-reverse justify-between py-8 px-20">
-                  {currentStep === Object.keys(formData).length ? (
-                    <Button type="submit" styleType={AppearanceStyleType.primary}>
-                      {t("t.submit")}
-                    </Button>
+                  {questionIndex === formData.length - 1 ? (
+                    <>
```
Comment (emilyjablonski) [inline sites/public/pages/finder.tsx:207]: 🧹

File: sites/public/pages/finder.tsx [hunk 9]
```diff
@@ -197,7 +230,7 @@ const Finder = () => {
                 </div>
 
                 <div className="flex justify-center align-center bg-white py-8">
-                  <a className="underline" href="/listings">
+                  <a className="underline" onClick={skipToListings}>
```
Comment (emilyjablonski) [inline sites/public/pages/finder.tsx:233]: We can't have an anchor tag without an href for a11y - we should use an anchor + href or a button + onClick

File: sites/public/pages/finder.tsx (hunk 6 above)
Comment (ColinBuyck) [reply sites/public/pages/finder.tsx:125]: Good point, @emilyjablonski. The only use cases I can think of come up with the disclaimer PR if we wanted to show all steps as completed. It's not currently on the Figma, but I believe I remember @slowbot mentioning that would be preferred if it wasn't too much additional work. The other one would be if we wanted to show skipped steps as empty, but don't think we've ever discussed this case.

File: sites/public/pages/finder.tsx (hunk 7 above)
Comment (ColinBuyck) [reply sites/public/pages/finder.tsx:180]: @emilyjablonski I included this to account for the names of the Regions.

File: sites/public/pages/finder.tsx (hunk 8 above)
Comment (ColinBuyck) [reply sites/public/pages/finder.tsx:201]: @emilyjablonski I found that rendering the submit button without a tag wrapping it resulted in it effectively being clicked which skips the disclaimer page. Do you know of a cleaner way around this issue?

**Comment by ColinBuyck** [comment]
Thank you for the review and good catches @emilyjablonski! I've updated the PR and responded to the comments that I had additional thoughts on. As for the functionality displayed in the screen recording, I believe that is intended based on the last bullet point of the AC.

File: sites/public/pages/finder.tsx (hunk 8 above)
Comment (ColinBuyck) [reply sites/public/pages/finder.tsx:201]: With some further research, I found this [discussion on it](https://stackoverflow.com/questions/71754898/involuntary-form-submit-when-rendering-submit-button-in-reactjs). Adding a key also solved the issue and prevented the auto-submit. Do you have a preference between these solutions?

File: sites/public/pages/finder.tsx (hunk 6 above)
Comment (ColinBuyck) [reply sites/public/pages/finder.tsx:125]: Update: Was working ahead on the disclaimer piece and setting the currentPageSection and completedSections to the same value results in the final step still showing as active. It only works with the currentPageSection set to one greater than the total number of steps so the completed step value is, once again, one less that the current 😮. So now the _only_ case I can think of is if we ever wanted something like the screenshot below.
image

Comment (emilyjablonski) [reply sites/public/pages/finder.tsx:125]: Sounds great thank you for the details! I think good to leave as is for now.

**Comment by emilyjablonski** [comment]
@ColinBuyck Thank you the changes look great! The only remaining thought is still on the video and I could definitely just have a different/wrong understanding of the AC, but I read the last bullet as if it's my first time through the form and I select some neighborhoods then hit back and return it won't save, but I read the second to last bullet as if I at some point select some and hit next, they should save even when hitting back if I return.

**Review by emilyjablonski — APPROVED** [review]
Chatted offline 😁

**Pull request merged**